File: frontend/src/config/plans.js
export const PLANS = [
  {
    id: "creator",
    name: "Creator",
    monthlyUsd: 19,
    yearlyUsd: 190,
    monthlyVideos: 27,
    monthlyClips: 12,
    monthlyMessages: 999,
    monthlyEmail: 999,
    monthlyOutlook: 999,
    monthlyInstagram: 60,
    monthlyX: 60,
    monthlyTiktok: 60,
    youtubeMinMinutes: 15,
    youtubeTargetMinutes: 25,
    description: "A focused operating system for creators building consistently.",
    couponDiscount: 50,
    features: [
      "27 YouTube long videos / month",
      "12 clips / month",
      "999 messages / month",
      "999 Gmail / Email / Outlook actions / month",
      "15–25+ minute YouTube videos",
      "AI content studio",
      "Production workflow",
      "Scheduling",
      "Creator analytics",
    ],
  },
  {
    id: "pro",
    name: "Pro",
    monthlyUsd: 49,
    yearlyUsd: 490,
    monthlyVideos: 72,
    monthlyClips: 39,
    monthlyMessages: 1999,
    monthlyEmail: 1999,
    monthlyOutlook: 1999,
    monthlyInstagram: 180,
    monthlyX: 180,
    monthlyTiktok: 180,
    youtubeMinMinutes: 15,
    youtubeTargetMinutes: 25,
    description: "For creators running a serious automated content operation.",
    couponDiscount: 25,
    popular: true,
    features: [
      "72 YouTube long videos / month",
      "39 clips / month",
      "1,999 messages / month",
      "1,999 Gmail / Email / Outlook actions / month",
      "15–25+ minute YouTube videos",
      "Everything in Creator",
      "Automation workflows",
      "Advanced insights",
      "Priority processing",
    ],
  },
  {
    id: "studio",
    name: "Studio",
    monthlyUsd: 129,
    yearlyUsd: 1290,
    monthlyVideos: 111,
    monthlyClips: 100,
    monthlyMessages: 4499,
    monthlyEmail: 4499,
    monthlyOutlook: 4499,
    monthlyInstagram: 360,
    monthlyX: 360,
    monthlyTiktok: 360,
    youtubeMinMinutes: 15,
    youtubeTargetMinutes: 25,
    description: "For high-volume creator operations and teams.",
    couponDiscount: 42,
    features: [
      "111 YouTube long videos / month",
      "100 clips / month",
      "4,499 messages / month",
      "4,499 Gmail / Email / Outlook actions / month",
      "15–25+ minute YouTube videos",
      "Everything in Pro",
      "High-volume generation",
      "Advanced operations",
      "Priority support",
    ],
  },
];

export const DEFAULT_CURRENCY = "USD";

export const YEARLY_SAVINGS = {
  creator: 38,
  pro: 98,
  studio: 258,
};

export function formatPlanPrice(
  plan,
  currency = "USD",
  rate = 1,
  yearly = false
) {
  const usd = yearly ? plan.yearlyUsd : plan.monthlyUsd;
  const amount = currency === "USD" ? usd : usd * rate;

  return new Intl.NumberFormat(undefined, {
    style: "currency",
    currency,
    maximumFractionDigits: currency === "USD" ? 0 : 2,
  }).format(amount);
}

File: backend/app/billing/plans.py
from dataclasses import dataclass


@dataclass(frozen=True)
class Plan:
    id: str
    name: str
    monthly_usd: int
    yearly_usd: int
    monthly_videos: int
    monthly_clips: int
    monthly_messages: int | None
    monthly_email: int | None
    monthly_outlook: int | None
    monthly_instagram: int
    monthly_x: int
    monthly_tiktok: int
    youtube_min_minutes: int
    youtube_target_minutes: int
    description: str


PLANS = {
    "creator": Plan(
        "creator",
        "Creator",
        19,
        190,
        27,
        12,
        999,
        999,
        999,
        60,
        60,
        60,
        15,
        25,
        "For creators building a consistent publishing system.",
    ),
    "pro": Plan(
        "pro",
        "Pro",
        49,
        490,
        72,
        39,
        1999,
        1999,
        1999,
        180,
        180,
        180,
        15,
        25,
        "For serious creators running an automated content operation.",
    ),
    "studio": Plan(
        "studio",
        "Studio",
        129,
        1290,
        111,
        100,
        4499,
        4499,
        4499,
        360,
        360,
        360,
        15,
        25,
        "For high-volume creators and creator operations.",
    ),
}


COUPONS = {
    "creator": {
        "discount_percent": 50,
        "code_env": "N1MOX_CREATOR_COUPON",
    },
    "pro": {
        "discount_percent": 25,
        "code_env": "N1MOX_PRO_COUPON",
    },
    "studio": {
        "discount_percent": 42,
        "code_env": "N1MOX_STUDIO_COUPON",
    },
}


def public_plans():
    return [
        {
            "id": plan.id,
            "name": plan.name,
            "monthly_usd": plan.monthly_usd,
            "yearly_usd": plan.yearly_usd,
            "monthly_videos": plan.monthly_videos,
            "monthly_clips": plan.monthly_clips,
            "monthly_messages": plan.monthly_messages,
            "monthly_email": plan.monthly_email,
            "monthly_outlook": plan.monthly_outlook,
            "monthly_instagram": plan.monthly_instagram,
            "monthly_x": plan.monthly_x,
            "monthly_tiktok": plan.monthly_tiktok,
            "youtube_min_minutes": plan.youtube_min_minutes,
            "youtube_target_minutes": plan.youtube_target_minutes,
            "description": plan.description,
            "coupon_discount_percent": COUPONS[plan.id]["discount_percent"],
        }
        for plan in PLANS.values()
    ]
